<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DataSourcePropForm</class>
 <widget class="QWidget" name="DataSourcePropForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>399</width>
    <height>1099</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Data Source Property</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_10">
   <item>
    <widget class="QComboBox" name="grid_blocks_comboBox">
     <item>
      <property name="text">
       <string>All Grid</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Blocks</string>
      </property>
     </item>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="block_groupBox">
     <property name="title">
      <string>Block</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <property name="leftMargin">
       <number>9</number>
      </property>
      <property name="topMargin">
       <number>9</number>
      </property>
      <property name="rightMargin">
       <number>9</number>
      </property>
      <property name="bottomMargin">
       <number>9</number>
      </property>
      <item>
       <widget class="QListWidget" name="block_listWidget"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="particles_groupBox">
     <property name="title">
      <string>particles</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <widget class="QLabel" name="label_17">
        <property name="text">
         <string>type</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="particlesType_comboBox">
        <item>
         <property name="text">
          <string>None</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Sphere</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="resolution">
        <property name="text">
         <string>Resolution</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="particlesRes_spinBox">
        <property name="minimum">
         <number>3</number>
        </property>
        <property name="maximum">
         <number>20</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Radius</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="particlesRadius_doubleSpinBox">
        <property name="decimals">
         <number>10</number>
        </property>
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.001000000000000</double>
        </property>
        <property name="value">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="lineType_groupBox">
     <property name="title">
      <string>Line Type</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_7">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Line Type</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="lineType_comboBox">
          <item>
           <property name="text">
            <string>Glyph</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Line</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTabWidget" name="lineType_tabWidget">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>Arrow</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QGridLayout" name="gridLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Tip resolution</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="tipRes_spinBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>128</number>
              </property>
              <property name="value">
               <number>6</number>
              </property>
              <property name="displayIntegerBase">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Tip radius</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QDoubleSpinBox" name="tipRad_doubleSpinBox">
              <property name="maximum">
               <double>1.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.010000000000000</double>
              </property>
              <property name="value">
               <double>0.080000000000000</double>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Tip length</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QDoubleSpinBox" name="tipLen_doubleSpinBox">
              <property name="maximum">
               <double>1.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.010000000000000</double>
              </property>
              <property name="value">
               <double>0.350000000000000</double>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Shaft resolution</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QSpinBox" name="shaftRes_spinBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>128</number>
              </property>
              <property name="value">
               <number>6</number>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Shaft radius</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QDoubleSpinBox" name="shaftRad_doubleSpinBox">
              <property name="maximum">
               <double>1.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.010000000000000</double>
              </property>
              <property name="value">
               <double>0.080000000000000</double>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QLabel" name="scaleFactor_label">
              <property name="text">
               <string>Scale Factor</string>
              </property>
             </widget>
            </item>
            <item row="5" column="1">
             <widget class="QDoubleSpinBox" name="scaleFactor_doubleSpinBox">
              <property name="decimals">
               <number>6</number>
              </property>
              <property name="maximum">
               <double>0.010000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.000010000000000</double>
              </property>
              <property name="value">
               <double>0.000200000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>Cone</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="0">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Resolution</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QSlider" name="Res_cone_horizontalSlider">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>10</number>
                </property>
                <property name="value">
                 <number>6</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="Res_cone_lineEdit">
                <property name="text">
                 <string>6</string>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>Radius</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QDoubleSpinBox" name="RadCone_doubleSpinBox">
                <property name="minimum">
                 <double>0.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>10.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.010000000000000</double>
                </property>
                <property name="value">
                 <double>0.100000000000000</double>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>Height</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="ConeHei_doubleSpinBox">
                <property name="maximum">
                 <double>10.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.010000000000000</double>
                </property>
                <property name="value">
                 <double>0.600000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="center_Cone_label">
              <property name="text">
               <string>Center</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QLineEdit" name="centerX_Cone_lineEdit">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="centerY_Cone_lineEdit">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="centerZ_Cone_lineEdit">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="direction_Cone_label">
              <property name="text">
               <string>Direction</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QLineEdit" name="DirX_lineEdit">
                <property name="text">
                 <string>1</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="DirY_lineEdit">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="DirZ_lineEdit">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>24</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_3">
         <attribute name="title">
          <string>Cylinder</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="0" column="0">
             <widget class="QLabel" name="label_15">
              <property name="text">
               <string>Tip resolution</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QSlider" name="Res_Cylinder_horizontalSlider">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>10</number>
                </property>
                <property name="singleStep">
                 <number>1</number>
                </property>
                <property name="value">
                 <number>6</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="Res_Cylinder_lineEdit">
                <property name="text">
                 <string>6</string>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_14">
              <property name="text">
               <string>Radius</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_7">
              <item>
               <widget class="QDoubleSpinBox" name="RadCylinder_doubleSpinBox">
                <property name="maximum">
                 <double>10.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.010000000000000</double>
                </property>
                <property name="value">
                 <double>0.100000000000000</double>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_12">
                <property name="text">
                 <string>Height</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="HeiCylinder_doubleSpinBox">
                <property name="minimum">
                 <double>0.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>10.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.010000000000000</double>
                </property>
                <property name="value">
                 <double>0.100000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="center_Cylinder_label">
              <property name="text">
               <string>Center</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <widget class="QLineEdit" name="CenterX_Cylinder_lineEdit">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="CenterY_Cylinder_lineEdit">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="CenterZ_Cylinder_lineEdit">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>52</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="seaBottom_groupBox">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="title">
         <string>Sea Bottom</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QCheckBox" name="seaBottom_checkBox">
           <property name="text">
            <string>Sea Bottom</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="seaBotColor_pushButton">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="displace_groupBox">
     <property name="title">
      <string>Displacement animation</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_9">
      <item>
       <widget class="QCheckBox" name="displace_checkBox">
        <property name="text">
         <string>Enabled</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="disPara_groupBox">
        <property name="title">
         <string/>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_8">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_9" stretch="8,2">
           <item>
            <widget class="QLineEdit" name="displace_lineEdit">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="displace_pushButton">
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="0" column="0">
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>Start position</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="pos_spinBox"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>Interval</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="interval_spinBox"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_16">
             <property name="text">
              <string>Zoom factor</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QDoubleSpinBox" name="zoomFactor_doubleSpinBox">
             <property name="minimum">
              <double>0.010000000000000</double>
             </property>
             <property name="maximum">
              <double>100.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QCheckBox" name="gravityCenter_checkBox">
           <property name="text">
            <string>Input center of gravity</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="gravity_groupBox">
           <property name="title">
            <string>Center of gravity</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_10">
            <item>
             <widget class="QLineEdit" name="gravityX_lineEdit"/>
            </item>
            <item>
             <widget class="QLineEdit" name="gravityY_lineEdit"/>
            </item>
            <item>
             <widget class="QLineEdit" name="gravityZ_lineEdit"/>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="comModel_groupBox">
     <property name="title">
      <string>combine model</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <item>
       <widget class="QCheckBox" name="comModel_checkBox">
        <property name="text">
         <string>combine model</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="comModel_comboBox"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
